app/model/command/CreatePillarCommand.scala

```diff
@@ -0,0 +1,64 @@
+package model.command
+
+import com.gu.tagmanagement.{PillarEvent, PillarEventType}
+import model._
+import play.api.libs.functional.syntax._
+import play.api.libs.json.{Format, JsPath}
+import repositories._
+import CommandError._
+import play.api.Logger
+import services.KinesisStreams
+
+import scala.concurrent.Future
+import services.Contexts.tagOperationContext
+
+case class CreatePillarCommand(
+  name: String,
+  path: String,
+  sectionIds: Seq[String] = Nil) extends Command {
+
+  type T = Pillar
+
+  def process()(implicit username: Option[String] = None): Future[Option[Pillar]] = {
+
+    val pageIdFuture: Future[Long] = Future { try { PathManager.registerPathAndGetPageId(path) } catch {
+      case p: PathRegistrationFailed => PathInUse
+      case e: Throwable =>
+        Logger.warn(s"Error using pathmanager for $path: ${e.getMessage}", e)
+        throw e
+    }}
+
+    val pillarIdFuture = Future{Sequences.pillarId.getNextId}
+
+    for {
+      pageId <- pageIdFuture
+      pillarId <- pillarIdFuture
+    } yield {
+      val pillar = Pillar(
+        id = pillarId,
+        name = name,
+        path = path,
+        pageId = pageId,
+        sectionIds = sectionIds
+      )
+
+      val result = PillarRepository.updatePillar(pillar)
+
+      KinesisStreams.pillarUpdateStream.publishUpdate(pillar.id.toString, PillarEvent(PillarEventType.Update, pillar.id, Some(Pillar.asThrift(pillar))))
+
+      PillarAuditRepository.upsertPillarAudit(PillarAudit.created(pillar))
+
+      result
+    }
+  }
+}
+
+object CreatePillarCommand {
+
+  implicit val createPillarCommandFormat: Format[CreatePillarCommand] = (
```

Comment:
Are we serializing the command?

Reply:
We deserialize in the controller, I suppose it doesn't need to be a `Format` but it's consistent with the rest of this app...

Reply:
Ah, apologies I just missed that :)